### DISCUSSION TITLE
Regarding dependencies.R, is it needed without rsconnect?

### DISCUSSION THREAD

**Discussion by SaintRod** [Appsilon/rhino#557] [Q&A] [answered]
In the dependencies.R file there is a comment that states:

```
# This file allows packrat (used by rsconnect during deployment) to pick up dependencies.
```

Questions:
1. Is this file needed if I'm not using rsconnect to deploy the app? Additionally, hasn't packrat been superseded by renv?
2. If this file is not needed, might a feature proposal be to give the user the option not to create this file?
3. What risk exists in deleting the file? Does anything depend on the file?

**Comment by kamilzyla** [comment, marked as answer]
Hi @SaintRod and thanks for asking!

Currently two features of Rhino use this file:
1. `pkg_install()` and `pkg_remove()` functions will recreate it, if it does not exist.
2.  Rhino uses a `.renvignore` file so that `renv` only reads dependencies only from `dependencies.R` ([details](https://appsilon.github.io/rhino/articles/explanation/renv-configuration.html)).

At the time when we devised this solution, `packrat` has been already deprecated for a while, but it was still being used by `rsconnect`  for dependency discovery (it only started using `renv` by default in version 1.0.0).

I'd like us to find a better solution, but it's not as simple as getting rid of `dependencies.R` and `.renvignore`, as we do want to have a central location with a list of all project dependencies (we tried using `DESCRIPTION` in the past, but we had some issues without using a package).

With that said, if you delete `dependencies.R` and `.renvignore`, your project should work fine, though I can't make a promise :wink:

**Reply by SaintRod** [reply]
Thanks for the insight!

PS. I'm developing a shiny app using Rhino for the first time and all I can say is well done! I've greatly enjoyed using Rhino.
